pylint/checkers/variables.py

```diff
@@ -1221,7 +1251,14 @@ def visit_name(self, node: nodes.Name) -> None:
                                 "undefined-variable", node=node, args=node.name
                             )
 
-            current_consumer.mark_as_consumed(node.name, found_nodes)
+            # Some nodes were filtered out because they execute only on success
+            # in try/finally or only on failure in except
+            # Add them back so that they are not unnecessarily marked as unused
+            try:
+                uncertain_nodes = current_consumer.consumed_uncertain[node.name]
+            except KeyError:
+                uncertain_nodes = []
+            current_consumer.mark_as_consumed(node.name, found_nodes + uncertain_nodes)
```

Comment:
```suggestion
            if node.name in current_consumer.consumed_uncertain:
                found_nodes  += current_consumer.consumed_uncertain[node.name]
            current_consumer.mark_as_consumed(node.name, found_nodes)
```

Little bit more concise and smaller diff.